code/modules/projectiles/guns/projectile/revolver.dm

```diff
@@ -252,6 +252,7 @@
 
 /obj/item/weapon/gun/projectile/revolver/russian/proc/shoot_self(mob/living/carbon/human/user, affecting = "head")
 	user.apply_damage(300, BRUTE, affecting)
+	user.do_suicide(BRUTE, "Russian Revolver")

```

Comment:
Did you test this? do_suicide doesn't set the mob's suiciding var to 1, so this doesn't prevent them from being cloned.